components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
+        const reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit
+        );
+        if (get(reddit_things, "data.after") !== undefined) {
+          after = reddit_things.data.after;
+          console.log(
+            `after children length: ${reddit_things.data.children.length}`
+          );
+          if (reddit_things.data.children.length > 0) {
+            reddit_things.data.children.forEach((reddit_link) => {
+              this.emitMeInfo(reddit_link);
+            });
+          }
+        }
+      } while (after);
+      this.db.set("after", after);
+      console.log(`finished [deploy] hook, after: ${after}`);
+    },
+  },
+  methods: {
+    emitMeInfo(reddit_link) {
+      this.$emit(reddit_link, {
+        summary: reddit_link.data.name,
+      });
+    },
+  },
+  async run() {
+    // Emmits new events as per timer config
+
+    let current_after = this.db.get("after");
+    console.log(`started [run] method, after: ${current_after}`);
+    const reddit_things = await this.reddit.getNewSubredditLinks(
+      current_after,
+      this.subreddit
+    );
+
+    if (get(reddit_things, "data.after") !== undefined) {
+      var new_after = reddit_things.data.after;
```

Comment:
If you implement pagination in the getNewSubredditLinks method, just make sure that you modify this to the max timestamp for all items returned across all pages.

Also change `var` to `let`.

Reply:
Ok, I decided to implement pagination in this method. However, I don't see how the timestamp is relevant here. The reddit objects returned by the API endpoint for this source don't have a field related to a last update/modify date (I am thinking in the strategy used in the sample mercury source with code like

```
 if (updatedAt.getTime() > updatedAfter.getTime()) {
          this.$emit(contact, this.generateMeta(contact, updatedAt));
        } else {
          // don't need to continue if we've gotten to contacts already evaluated
          done = true;
        }
```

As per reddit API documentation, for pagination only `after` or `before` are given (optionally use a `count` to keep truck of the total number of records having been pulled)

> To page through a listing, start by fetching the first page without specifying values for after and count. The response will contain an after value which you can pass in the next request. It is a good idea, but not required, to send an updated value for count which should be the number of items already fetched.

Could you elaborate on the use of the timestamp? I can think it can be included in the metadata for the event to indicate when the event is being emitted

Reply:
I moved this question to a newer one addressing timestamps.